UI Tests: Enable screenshots in CI

## Description
Cypress UI Tests can generate screenshots for failing tests that are helpful for debugging.  These should be pushed to storage and viewable by authors of PR's.

## Expected Behavior
- [ ] Screenshots are enabled
- [ ] Screenshots are pushed to Storage Bucket
- [ ] Links to screenshots are surfaced in logs
- [ ] Screenshots are deleted on VM or CI  has sufficient permission to delete screenshots on next build

## Current Behavior 
Screenshots are currently disabled because cypress docker container runs as root, resulting in artifacts that only root user can delete.  When CI re-runs, it is unable to delete the screenshots directory from the previous build.

```
##[error]One or more errors occurred. (One or more errors occurred. (Access to the path '/home/user/actions-runner/_work/anthos-finance-demo/anthos-finance-demo/e2e/cypress/screenshots/login_spec.js/Bad Credentials on Form Submission -- fails with alert banner (failed).png' is denied.)) (Access to the path '/home/user/actions-runner/_work/anthos-finance-demo/anthos-finance-demo/e2e/cypress/screenshots/login_spec.js/Bad Credentials on Form Submission -- fails with alert banner (failed).png' is denied.)
##[error]Exit code 1 returned from process: file name '/home/user/actions-runner/bin.2.168.0/Runner.PluginHost', arguments 'action "GitHub.Runner.Plugins.Repository.v1_0.CheckoutTask, Runner.Plugins"'.
```

## Possible solutions
- Run cypress as non-root
- Modify permissions on screenshots/ dir
- Move screenshots out of _work dir (temp fix, would need to be cleaned up manually)

--- a/.github/workflows/ci-master.yaml
+++ b/.github/workflows/ci-master.yaml
@@ -157,4 +157,9 @@ jobs:
         sleep 60
     - name: Run UI tests (Cypress)
       run: |
-        docker run -t -v $PWD/.github/workflows/ui-tests:/e2e -w /e2e -e CYPRESS_baseUrl=http://$EXTERNAL_IP --network=host cypress/included:5.0.0 --env CI=true
+        docker run -t -v $PWD/.github/workflows/ui-tests:/e2e -w /e2e -e CYPRESS_baseUrl=http://$EXTERNAL_IP -u node cypress/included:5.0.0 
+    - uses: actions/upload-artifact@v2
+      if: failure()
+      with:
+          name: cypress-screenshots
+          path: ${{ github.workspace }}/.github/workflows/ui-tests/cypress/screenshots

--- a/.github/workflows/ui-tests/README.md
+++ b/.github/workflows/ui-tests/README.md
@@ -51,7 +51,7 @@ make test-e2e E2E_FLAGS="--spec cypress/integration/<test-file>.js"
 ```console
 docker run -it -v ${E2E_PATH}:/e2e -w /e2e \
  -e CYPRESS_baseUrl=${E2E_URL} -e CYPRESS_CI=false \
- cypress/included:4.3.0 --spec cypress/integration/<test-file>.js
+ cypress/included:5.0.0 --spec cypress/integration/<test-file>.js
 ```
 
 ### Filtering tests with `.only()` and `.skip()`
@@ -60,4 +60,4 @@ The `.only()` and `.skip()` functions can be appended to either single tests or 
 
 ### Screenshots
 
-Screenshots are enabled by passing the environment variable `CI=false`.  When running tests locally, failing tests will store screenshots in the `cypress/screenshots` directory. These are helpful when debugging tests.  [Read more about them here](https://docs.cypress.io/guides/guides/screenshots-and-videos.html#Screenshots).
+Failing tests will store screenshots in the `cypress/screenshots` directory (locally and CI) and uploaded as an [artifact to Github Actions](https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts) after a failing build. These are helpful when debugging tests.  [Read more about them here](https://docs.cypress.io/guides/guides/screenshots-and-videos.html#Screenshots).

